chainer/links/activation/dropconnect.py

```diff
@@ -0,0 +1,94 @@
+import math
+
+from chainer import cuda
+from chainer.functions.noise import dropconnect
+from chainer import initializers
+from chainer import link
+
+
+class Dropconnect(link.Link):
+
+    """Fully-connected layer with dropconnect regularization.
+
+    Args:
+        in_size (int): Dimension of input vectors. If None, parameter
+            initialization will be deferred until the first forward data pass
+            at which time the size will be determined.
+        out_size (int): Dimension of output vectors.
+        wscale (float): Scaling factor of the weight matrix.
+        initialW (3-D array or None): Initial weight value.
+            If ``None``, then this function uses ``wscale`` to initialize.
+        initial_bias (2-D array, float or None): Initial bias value.
+            If it is float, initial bias is filled with this value.
+            If it is ``None``, bias is omitted.
+        reuse_mask (boolean):
+            If ``True``, once dropconnect mask is generated,
+            same mask will be reuse when link is called.
```

Comment:
So far as I know, users usually do not want `dropconnect` link to use same mask throughout the training. I would like to know if there is a usecase in which users want to fix the mask throughout whole training procedure. If there weren't, I think it is better not to add this option, considering the balance of the complexity of API and benefit for users.

On the other hand, I agree with `debug_mask` option in the `dropout` function because user may want to use this option for debugging.

So far as I check the implementation, `reuse_mask` is used to pass `check_backward` test, in which we call the link several times to calculate numerical gradient. So, if we could test the behavior of backward of the link, we may have to add this argument. Could you consider the feasibility of this proposal? One of my ideas is to set mask manually (or through the APIs of `dropconnect` function) in calculating the numerical grad.